src/core/api/index.js

```diff
@@ -308,3 +308,19 @@ export function logOutFromServer({ api }: {| api: ApiStateType |}) {
     state: api,
   });
 }
+
+type AutocompleteParams = {|
+  api: ApiStateType,
+  query: string,
+|};
+
+export function autocomplete({ api, query }: AutocompleteParams) {
+  return callApi({
+    endpoint: 'addons/autocomplete',
+    params: {
```

Comment:
The [autocomplete API](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#get--api-v3-addons-autocomplete-) also supports `addonType` which we should set if the `viewContext` is `ADDON_TYPE_EXTENSION`, `ADDON_TYPE_THEME`, etc. The intent of the "sections" of the site "Explore, Extensions, Themes, etc." is that searches are scoped to that add-on type when you're in that section (at least by default). So only when on the homepage/"Explore" should you search all add-on types.

Also we have a bug open for it for regular search but we might as well use `appversion` here so only compatible add-ons are returned.